### ISSUE TITLE
`import/order`: Impose custom ordering within groups

### ISSUE THREAD

**Issue by futpib** [benmosher/eslint-plugin-import#1378]
For example, new option could look like this:
```js
{
	"withinGroups": {
		"external": [
			"ramda",
			"react"
		]
	}
}
```

With this option, this would pass:

```js
import { compose } from 'ramda';
import somethingElse from 'something-else'; // could be placed anywhere within the "external" group
import React from 'react';
```

And this would fail:

```js
import React from 'react';
import { compose } from 'ramda';
```

**Comment by ljharb** [comment]
Why? What’s the reason you’d want ordering, but also an arbitrary order?

**Comment by futpib** [comment]
I primarily want ordering, arbitrary order only as a fallback for imports from packages for which the order is not configured. Another option is to report such imports to force users to update their configuration.

**Comment by kyranjamie** [comment]
Came here to second this. Ordering alphabetical means very little to me, personally. I want to group my domain/purpose. Beginning with framework, for example. An Angular app, `@angular/*` goes first, react then `react` etc

**Comment by billyvg** [comment]
This is something I'm looking for so that we can also enforce new lines between custom groups.

**Comment by kevinwolfdev** [comment]
I think #1746 is related to this.

**Comment by amannn** [comment]
I have a use case that might be related to this.

```js
import React from 'react';
import Button from 'components/Button';
import config from './config';
import styles from './ErrorBoundaryError.scss';
```

In this case `import/order` wants to put `styles` before `config`.

I'd like to always have imports to a local stylesheet at the very bottom. Ideally I'd be able to match files based on a glob within a group and move them to the bottom.

**Comment by tyteen4a03** [comment]
Bump - would be very useful :)
